main.tf

```diff
@@ -103,9 +106,9 @@ module "lacework_at_svc_account" {
 }
 
 resource "google_storage_bucket" "lacework_bucket" {
-  count                       = length(var.existing_bucket_name) > 0 ? 0 : 1
+  count                       = local.create_storage_bucket ? 1 : 0
   project                     = local.project_id
-  name                        = "${var.prefix}-${random_id.uniq.hex}"
+  name                        = coalesce(var.existing_bucket_name, "${var.prefix}-${random_id.uniq.hex}")
```

Comment:
If `create_storage_bucket` is true and `existing_bucket_name` is set, would this logic try to "create" the existing bucket (causing a failure)?

Reply:
Yes that is exactly what it does. Of course, if there is an existing bucket with this name configured outside of this terraform it will cause a (non-destructive) EntityAlreadyExists error.

Reply:
As an alternative, possibly less confusing, we could add a variable ‘var.custom_bucket_name’. You wouldn’t have to use ‘existing_bucket_name’ and the ‘var.create_storage_bucket’ is not necessary. Behavior when both are set should be decided on, a validation could help.

Reply:
A third approach would be to refactor the module and match the behavior in your other module [terraform-gcp-config](https://github.com/lacework/terraform-gcp-config/blob/7fc38579f36a78f89a55d422b2b4a91708334936/main.tf#L6-L10), though that would be a breaking change.